lib/project_types/script/layers/infrastructure/assemblyscript_dependency_manager.rb

```diff
@@ -34,9 +34,10 @@ def check_node_version!
           output, status = @ctx.capture2e("node", "--version")
           raise Errors::DependencyInstallError, output unless status.success?
 
-          version = Semantic::Version.new(output[1..-1])
-          unless version >= Semantic::Version.new("12.16.0")
-            raise Errors::DependencyInstallError, "Node version must be >= v12.16.0 Current version: #{output}."
+          require 'semantic/semantic'
```

Comment:
Is it standard to put the require here instead of at the top of the file?

Reply:
Standard no, but it's definitely a technique used to lazy load dependencies only when you actually need it.

Reply:
Yes it is more ideal to do it where needed in the CLI so that startup is quicker.

Reply:
Makes sense.